htdocs/js/notebook/notebook_model.js

```diff
@@ -35,14 +35,25 @@ Notebook.create_model = function()
                 return asset.filename() == filename;
             });
         },
-        append_asset: function(asset_model, filename, skip_event) {
+        append_asset: function(asset_model, filename, skip_event, user_appended) {
             asset_model.parent_model = this;
             var changes = [];
             changes.push(asset_model.change_object());
-            this.assets.push(asset_model);
+            var asset_filenames = _.map(this.assets, function(asset) { return asset.filename(); });
+            var new_asset_index;
+           
+            if(user_appended){
+                asset_filenames.push(asset_model.change_object().filename);
+                asset_filenames.sort((a,b) => a.localeCompare(b, undefined, {sensitivity: 'base'}));
+                new_asset_index =  asset_filenames.indexOf(asset_model.change_object().filename);
+                this.assets.splice(new_asset_index, 0, asset_model);
```

Comment:
This wouldn't work unless they were already sorted (because both arrays have to have the same order), so I don't think the sort here is actually doing anything.

Since the assets are already in alphabetical order, it shouldn't be necessary to create a secondary array of all the filenames. 

You should be able to directly find the index of the first asset whose name is greater than the filename of the asset to be inserted, say using [Array.findIndex](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) and the same `localeCompare` operation.

Reply:
I feel silly making you return to this after neglecting this for so long, so I'll make the above change when I test & merge tomorrow.